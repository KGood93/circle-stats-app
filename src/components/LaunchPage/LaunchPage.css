.LaunchPage {
    padding-bottom: 275px;
}

.launchHead {
    font-family: 'Roboto', sans-serif;
    color: rgb(12, 70, 94);
}

.mainButton {
    height: 40px;
    background-color: rgb(12, 70, 94);
    border: solid 2px rgb(246, 125, 72);
    margin-top: 10px;
}

.mainButton a {
    vertical-align: middle;
    text-decoration: none;
    font-family: 'Press Start 2P', cursive;
    color: rgb(240, 198, 90);
}