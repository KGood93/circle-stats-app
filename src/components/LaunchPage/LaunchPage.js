import React, {Component} from 'react'
import {Link} from 'react-router-dom'
import config from '../../config'

class LaunchPage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            stats: [],
        }
    }

    componentDidMount() {
        this.fetchStats()
    }

    fetchStats() {
        const statsUrl = `${config.API_ENDPOINT}/stats`
        const statsOptions = {
          method: 'GET', 
          headers: {
            'Content-type': 'application/json'
          }
        }
    
        fetch(statsUrl, statsOptions)
          .then(res => {
            if(res.ok) {
              return res.json()
            }
            else {
              throw new Error('Something went wrong loading stats')
            }
          })
          .then(data => {
            //console.log(data)
            this.setState({stats: data})
          })
          .catch(err => {
            this.setState({
              error: err.message
            })
          })
      }

    getMeetId() {
        const {meet_id} = this.props.match.params
        const id = parseInt(meet_id)
        return id
    }

    renderLastMessage() {
        const stat = this.state.stats
        //console.log(stat)

        const meetupId = this.getMeetId()

        const meetStat = []

        for(var i = 0; i < stat.length; i++) {
            if(stat[i].meet_id === meetupId) {
                meetStat.push(stat[i])
            }
        }

        if(meetStat.length === 0) {
            return(
                <div className="welcomeMessage">
                    <h2 className="launchHead">This is your first meetup!</h2> 
                </div>
            )
        }
        else {
            let dataIndex = meetStat.length - 1
            //console.log(dataIndex)
            let meetup = meetStat[dataIndex]

            return(
                <div className="welcomeMessage">
                    <h2 className="launchHead">Your last meetup at {meetup.location} had</h2>
                    <h2 className="launchHead">{meetup.attendance}</h2>  
                    <h2 className="launchHead">attendees!</h2> 
                </div>
            )
        }

    }

    render() {
        return(
            <div className="LaunchPage">
                {this.renderLastMessage()}
                <div className="launchButtons">
                    <button type="button" className="mainButton"><Link to={`/doorCount/${this.getMeetId()}`}>Create New</Link></button>
                    <br/>
                    <button type="button" className="mainButton"><Link to={`/stats/${this.getMeetId()}`}>View Stats</Link></button>
                </div>
            </div>
        )
    }
}

export default LaunchPage