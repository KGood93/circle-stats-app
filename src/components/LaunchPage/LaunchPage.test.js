import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import LaunchPage from './LaunchPage';

describe('LaunchPage Component', () => {
    it(`renders without crashing`, () => {
        const test = {
            path: "/launch/:meet_id",
            url: "/launch/1",
            isExact: true,
            params: {meet_id: "1"}
        };
        const wrapper = shallow(<LaunchPage match={test}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});