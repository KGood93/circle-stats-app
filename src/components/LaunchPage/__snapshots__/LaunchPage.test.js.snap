// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchPage Component renders without crashing 1`] = `
<div
  className="LaunchPage"
>
  <div
    className="welcomeMessage"
  >
    <h2
      className="launchHead"
    >
      This is your first meetup!
    </h2>
  </div>
  <div
    className="launchButtons"
  >
    <button
      className="mainButton"
      type="button"
    >
      <Link
        to="/doorCount/1"
      >
        Create New
      </Link>
    </button>
    <br />
    <button
      className="mainButton"
      type="button"
    >
      <Link
        to="/stats/1"
      >
        View Stats
      </Link>
    </button>
  </div>
</div>
`;
